src/common-crypto/CryptoBackend.ts

```diff
@@ -106,11 +106,11 @@ export interface SyncCryptoCallbacks {
     preprocessToDeviceMessages(events: IToDeviceEvent[]): Promise<IToDeviceEvent[]>;
 
     /**
-     * Called by the /sync loop whenever there are incoming to-device messages.
+     * Called by the /sync loop when one time keys and unused fallback keys are received.
      *
-
-     * Note that, unlike {@link ClientEvent.ToDeviceEvent} events, this is called on the raw to-device
-     * messages, rather than the results of any decryption attempts.
+     * The presence of device_unused_fallback_key_types indicates that the
+     * server supports fallback keys. If there's no unused
+     * signed_curve25519 fallback key we need a new one.
```

Comment:
sorry, I guess I'm not making myself clear. I think this is an implementation note that belongs with the code that you have moved from `sync.ts` to `crypto/index.ts`.

It should be here: https://github.com/matrix-org/matrix-js-sdk/pull/3226/files#diff-890ab04f8e5e8b612d0069bf0c681ef35d1ef3810abba98205ee464f3c366577R3237, possibly with some slight clarification:

```javascript
        if (unusedFallbackKeys) {
            // If `unusedFallbackKeys` is truthy, that means `device_unused_fallback_key_types`
            // is present in the sync response, which indicates that the server supports fallback keys. 
            // 
            // If there's no unused signed_curve25519 fallback key, we need a new one.
            this.setNeedsNewFallback(!unusedFallbackKeys.includes("signed_curve25519"));
        }
```